app/webhook.go

```diff
@@ -789,10 +789,29 @@ func (a *App) HandleIncomingWebhook(c *request.Context, hookID string, req *mode
 		overrideIconURL = req.IconURL
 	}
 
-	_, err := a.CreateWebhookPost(c, hook.UserId, channel, text, overrideUsername, overrideIconURL, req.IconEmoji, req.Props, webhookType, "")
+	postingUserID := a.GetBotUserId(hook.UserId, req.Username)
+	if postingUserID == nil {
+		postingUserID = &hook.UserId
+	}
```

Comment:
Given the other change

```suggestion
	if postingUserID == "" {
		postingUserID = hook.UserId
	}
```

Reply:
Done

Reply:
cc: @mickmister @hanzei